pipeline/inferrer/feature_inferrer/app/src/batching.py

```diff
@@ -0,0 +1,119 @@
+import asyncio
+from uuid import UUID, uuid4
+from typing import Callable, Dict, Generic, List, TypeVar
+from async_timeout import timeout
+from src.logging import get_logstash_logger
+
+
+logger = get_logstash_logger(__name__)
+
+T = TypeVar("T")
+R = TypeVar("R")
```

Comment:
not sure what the significance of `T` and `R` are here... could/should they be more descriptive in some way?

Reply:
Convention in languages with generics, but I'll change them as it's not a particularly helpful convention